Source/SwiftLintBuiltInRules/Rules/Style/FunctionNameWhitespaceRule.swift

```diff
@@ -0,0 +1,67 @@
+import SwiftSyntax
+
+@SwiftSyntaxRule(correctable: true, optIn: true)
+struct FunctionNameWhitespaceRule: Rule {
+    var configuration = SeverityConfiguration<Self>(.warning)
+
+    static let description = RuleDescription(
+        identifier: "function_name_whitespace",
+        name: "Function Name Whitespace",
+        description: "Function declarations must have exactly one space before the name and no space after it",
+        kind: .style,
+        nonTriggeringExamples: [
+            Example("func abc(lhs: Int, rhs: Int) -> Int {}"),
+        ],
+        triggeringExamples: [
+            Example("func  name(lhs: A, rhs: A) -> A {}"),       // 2 spaces before
+            Example("func ↓name (lhs: A, rhs: A) -> A {}"),      // 1 space after
+            Example("↓func  ↓name (lhs: A, rhs: A) -> A {}"),    // 1 space after, 2 spaces before
+        ],
+        corrections: [
+            Example("↓func  ↓name (lhs: A, rhs: A) -> A {}"): Example("func name(lhs: A, rhs: A) -> A {}"),
+            Example("↓func  name(lhs: A, rhs: A) -> A {}"): Example("func name(lhs: A, rhs: A) -> A {}"),
+            Example("↓func   name(lhs: A, rhs: A) -> A {}"): Example("func name(lhs: A, rhs: A) -> A {}"),
+            Example("func ↓name (lhs: A, rhs: A) -> A {}"): Example("func name(lhs: A, rhs: A) -> A {}"),
+        ]
+    )
+}
+
+private extension FunctionNameWhitespaceRule {
+    final class Visitor: ViolationsSyntaxVisitor<ConfigurationType> {
+        override func visitPost(_ node: FunctionDeclSyntax) {
+            guard node.isNamedFunction else { return }
+            if !node.funcKeyword.trailingTrivia.isSingleSpace {
+                violations.append(
+                    at: node.funcKeyword.positionAfterSkippingLeadingTrivia,
+                    correction: ReasonedRuleViolation.ViolationCorrection(
+                        start: node.funcKeyword.endPositionBeforeTrailingTrivia,
+                        end: node.name.positionAfterSkippingLeadingTrivia,
+                        replacement: " "
+                    )
+                )
+            }
+            if node.name.trailingTrivia.isNotEmpty {
```

Comment:
A new implementation should probably accommodate generic parameters as well.

Reply:
Thanks! Generic parameters are parsed separately and not included in `trailingTrivia`, so the rule still behaves correctly. I added examples with generics to both `triggeringExamples` and `corrections` to ensure full coverage.

Let me know if you had a specific case in mind!

Reply:
I was rather thinking about something like `func f<T> ()`.